sklearn/datasets/_samples_generator.py

```diff
@@ -677,8 +688,16 @@ def make_moons(n_samples=100, shuffle=True, noise=None, random_state=None):
         The integer labels (0 or 1) for class membership of each sample.
     """
 
-    n_samples_out = n_samples // 2
-    n_samples_in = n_samples - n_samples_out
+    if isinstance(n_samples, (int, np.integer)):
+        n_samples_out = n_samples // 2
+        n_samples_in = n_samples - n_samples_out
+    else:
+        try:
+            n_samples_out, n_samples_in = n_samples
+        except ValueError:
+            raise ValueError('`n_samples` can be either an int or '
+                             'a two-element tuple. You passed ' +
+                             str(n_samples) + ' instead.')
```

Comment:
let's remove str here to keep consistent with make_circles

Reply:
Sure, I am just wondering why this was not caught by the tests (expected strings in unit tests were changed).